contracts/core/modules/RebalancingSetIssuanceModule.sol

```diff
@@ -364,9 +343,75 @@ contract RebalancingSetIssuanceModule is
 
     /* ============ Private Functions ============ */
 
+    /**
+     * Gets base set address from rebalancing set token and calculates amount of base set needed
+     * for issuance.
+     *
+     * @param  _rebalancingSetAddress    Address of the RebalancingSetToken to issue
+     * @param  _rebalancingSetQuantity   The Quantity of the rebalancing SetToken to issue
+     * @return baseSetAddress            The address of RebalancingSet's base SetToken
+     * @return requiredBaseSetQuantity   The quantity of base SetToken to issue
+     */
+    function getBaseSetAddressAndQuantity(
+        address _rebalancingSetAddress,
+        uint256 _rebalancingSetQuantity
+    )
+        internal
+        view
+        returns (address, uint256)
+    {
+        // Validate the rebalancing SetToken is valid and the quantity is a multiple of the natural unit
+        validateRebalancingSetIssuance(_rebalancingSetAddress, _rebalancingSetQuantity);
+
+        address baseSetAddress = IRebalancingSetTokenV2(_rebalancingSetAddress).currentSet();
+
+        // Calculate required base SetToken quantity
+        uint256 requiredBaseSetQuantity = getBaseSetIssuanceRequiredQuantity(
+            _rebalancingSetAddress,
+            _rebalancingSetQuantity
+        );
+
+        return (baseSetAddress, requiredBaseSetQuantity);        
+    }
+
+    /**
+     * Issues the rebalancing set token to sender and returns any excess baseSet.
+     *
+     * @param  _rebalancingSetAddress    Address of the rebalancing SetToken to issue
+     * @param  _baseSetAddress           Address of the rebalancing SetToken's base set
+     * @param  _rebalancingSetQuantity   The Quantity of the rebalancing SetToken to redeem
+     * @param  _keepChangeInVault        Boolean signifying whether excess base SetToken is transferred to the user 
+     *                                   or left in the vault
+     */
+    function issueRebalancingSetAndReturnExcessBase(
+        address _rebalancingSetAddress,
+        address _baseSetAddress,
+        uint256 _rebalancingSetQuantity,
+        bool _keepChangeInVault
+    )
+        internal
+    {
+        // Issue rebalancing SetToken to the sender
+        coreInstance.issueTo(
+            msg.sender,
```

Comment:
This internal function depends on msg.sender as implicit parameter, which makes code harder to read.  Consider making tthis parameter explicit.